.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,32 @@
+<!--
```

Comment:
I don't think that we need this.
It adds a lot of boilerplate that we haven't needed to date.
I think that a better requirement is to ensure that an issue has been created that clearly states the work, and that the PR description describes what has been done.

Our policy is that the PR description will be used as the comment on the final squashed commit. PRs often change, with features removed or added over the course of the review, and the PR description should be updated to reflect the changes.

Reply:
This works well if there has been out of band (Slack) communication about the PR, but maybe not well if someone wants to provide a bit more motivation and context than a dry change summary. I've added a "change summary" section to the template, and an instruction that this should be according to current policy. Upon merge, just this section can be used for the commit message.

Reply:
My main concern is that our current method is well established, and there does not appear to be a strong reason for making the change. Templates are usually added to solve a problem (e.g. we observe that PRs have incomplete or incorrect messages), and we should wait until we observe problems that need solving, then discuss solutions like templates.

The current approach allows committers to add context. The issue associated with the PR can also provide context.

Reply:
The strong reason is that external contributors have no idea about our methods. I have added a line to Contributing.md, because you are correct to point out it was not there.

The idea is that it helps us by helping contributors discover CONTRIBUTING exists, and layout the PR message. I've condensed the template slightly: what we care about is motivation and the change summary, and it's helpful that they are separated. We could condense the checklist maybe, but I'd like to keep the link to CONTRIBUTING somewhere.

Reply:
We don't have external collaborators knocking our doors down to make PRs, so we don't know what the problems are with their PRs. So how about spelling out our requirements for contributing in the `CONTRIBUTIONS.md` file, and if we are lucky enough to have external contributions, they can be pointed there if their PR isn't up to scratch.